components/ome-xml/src/ome/units/UNITS.java

```diff
@@ -66,7 +66,7 @@ public String getName()
   public static final ome.units.unit.Unit<ElectricPotential> VOLT    = ome.units.unit.Unit.<ElectricPotential>CreateBaseUnit("SI.VOLT", "V");
   public static final ome.units.unit.Unit<Frequency>         HERTZ   = ome.units.unit.Unit.<Frequency>CreateBaseUnit("SI.HERTZ", "Hz");
   public static final ome.units.unit.Unit<Length>            METRE   = ome.units.unit.Unit.<Length>CreateBaseUnit("SI.METRE", "m");
-  public static final ome.units.unit.Unit<Length>            INCH    = ome.units.unit.Unit.<Length>CreateBaseUnit("Imperial.INCH", "in");
+  public static final ome.units.unit.Unit<Length>            INCH    = METRE.multiply(0.0254).setSymbol("in");
```

Comment:
MeasurementSystem has to be immutable. It is what governs whether units are convertible.  If you did `setMeasurementSystem("Imperial")` the resulting object would have the same problem as the one created by `CreateBaseUnit("Imperial.INCH", "in");` so you would be back to writing custom code to handle metric<->imperial in `Length.java`.